cpp/src/cider/exec/nextgen/jitlib/llvmjit/LLVMJITUtils.h

```diff
@@ -87,9 +88,12 @@ inline llvm::Value* getLLVMConstantInt(NativeType value, llvm::LLVMContext& ctx)
     case JITTypeTag::INT64:
       return llvm::ConstantInt::get(type, (uint64_t)value, true);
     case JITTypeTag::INT128:
-      return llvm::ConstantInt::get(
-          ctx,
-          ((llvm::APInt(128, value >> 64) << 64) | llvm::APInt(128, (uint64_t)value)));
+      if constexpr (std::is_same_v<NativeType,
+                                   JITTypeTraits<JITTypeTag::INT128>::NativeType>) {
```

Comment:
when is it false?

Reply:
all types instead of ```INT128```, just to avoid a compiler warning.

Reply:
but this path only handles the `INT128`.
Does the compiler emit the warning for `value >> 64`?

Reply:
Yes.

Reply:
why wrapped in `const if` can suppress the warning?

Reply:
The code block is omitted in template instantiation.